### PR TITLE
Update solidity-coverage to v0.7

### LINKED ISSUE #86: Add coverage measure tool
We may want to use `solcover` like we are doing in many other repos

### PR DESCRIPTION
**Pull request by bingen**
Fixes #86

### CODE REVIEW

**Comment by facuspagnuolo** [comment]
I don't see a CI job for this, do you know why?

File: .travis.yml [hunk 1]
```diff
@@ -43,3 +43,9 @@ jobs:
     - stage: tests
       script: npm test test/treasury/court-treasury.js
       name: "Treasury"
+    - stage: tests
+      script npm run coverage
```
Comment (sohkai) [inline .travis.yml]:
```suggestion
    - stage: coverage
      script: npm run coverage
```

File: test/lib/hex-sum-tree-gas.js [hunk 2]
```diff
@@ -24,7 +24,7 @@ contract('HexSumTree', () => {
     })
   }
 
-  describe('gas costs', () => {
+  describe('gas costs [ @skip-on-coverage ]', () => {
```
Comment (sohkai) [inline test/lib/hex-sum-tree-gas.js]: This syntax is super super nice!

Do these fail though? solidity-coverage@0.7 shouldn't be increasing gas costs anymore, right?

File: .travis.yml (hunk 1 above)
Comment (facuspagnuolo) [reply .travis.yml]: this must be the reason why the CI didn't trigger a job for this PR... applying this suggestion to start the CI and see how it goes

File: test/lib/hex-sum-tree-gas.js (hunk 2 above)
Comment (facuspagnuolo) [reply test/lib/hex-sum-tree-gas.js]: interesting, let's give it a try

File: .travis.yml (hunk 1 above)
Comment (bingen) [reply .travis.yml]: Why wouldn’t it trigger the job? It’s the same stage as others. :thinking: 
It’s funny that I initially set this as `stage: coverage`, but then I saw that `lint` was also on `stage: tests` so I changed it to be consistent (I see coverage more like tests than linting).

File: test/lib/hex-sum-tree-gas.js (hunk 2 above)
Comment (bingen) [reply test/lib/hex-sum-tree-gas.js]: Yes, these were failing. Actually  I took that nice syntax from `solidity-coverage` documentation when I realized gas tests were failing.

File: .travis.yml (hunk 1 above)
Comment (bingen) [reply .travis.yml]: Oh, the `:`!!! :man_facepalming:

**Comment by facuspagnuolo** [comment]
For some reason the `@skip-coverage` notation is not working

**Comment by bingen** [comment]
> For some reason the @skip-coverage notation is not working

Do you mean in the CI? Because in locally it was working for me. I don't see gas tests in the coverage task either. Where are you seeing them?

**Review by facuspagnuolo — COMMENTED** [review]
Let's see how we can handle the travis timeout, not sure if there is a way to instrument the code in batches

**Comment by bingen** [comment]
> Let's see how we can handle the travis timeout, not sure if there is a way to instrument the code in batches

Being coverage I don't know how we can batch it. What I would do for now is to remove the travis job and merge this PR, so at least we can run coverage locally.

**Comment by sohkai** [comment]
> Being coverage I don't know how we can batch it. What I would do for now is to remove the travis job and merge this PR, so at least we can run coverage locally.

We could do a quick experiment or two using lcov-merger (see aragon-apps) if we separate coverage runs to test separate files from each other.

**Comment by facuspagnuolo** [comment]
Closing this for now

**Pull request closed without merging**